<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>ICCE User Guide</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">ICCE User Guide</h1>



<p>ICCE can process Illumina Infinium DNA methylation data. It currently only supports the EPIC Platform.</p>
<div id="install-icce" class="section level1">
<h1>Install ICCE</h1>
<p>Development version can be installed from github</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&#39;ethanmoyer/icce&#39;</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a>BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&#39;ethanmoyer/icceData&#39;</span>)</span></code></pre></div>
</div>
<div id="creating-consensus-vectors-with-the-ice-pipeline" class="section level1">
<h1>Creating consensus vectors with the ICE Pipeline</h1>
<p>The ICCE data package contains a small reference file that can be obtained through:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a>ref_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/reference.rds&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;sesameData&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a>reference &lt;-<span class="st"> </span><span class="kw">get_reference</span>(ref_file)</span></code></pre></div>
<p>ICCE utilizes the openSesame pipeline through:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a>idat_dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;sesameData&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true"></a>betas &lt;-<span class="st"> </span><span class="kw">read_IDAT</span>(idat_dir)</span></code></pre></div>
<p>where <code>idat_dir</code> is the directory containing all the IDAT files (they can be present under nested sub-directories).</p>
<p>Some data may have groups that are heterogeneous or undesirable for tree reconstruction. These can be removed by simply including their sample names in the following variable:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a>remove_names =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;PCA0612&quot;</span>, <span class="st">&quot;WB1148&quot;</span>, <span class="st">&quot;A2.mix&quot;</span>, <span class="st">&quot;B6.mix&quot;</span>, <span class="st">&quot;PCA1547&quot;</span>, <span class="st">&quot;WB1147&quot;</span>, <span class="st">&quot;B3.mix&quot;</span>, <span class="st">&quot;B2.mix&quot;</span>, <span class="st">&quot;A4.mix&quot;</span>, <span class="st">&quot;B4.mix&quot;</span>, <span class="st">&quot;A3.mix&quot;</span>, <span class="st">&quot;ST2007&quot;</span>, <span class="st">&quot;B5.mix&quot;</span>, <span class="st">&quot;B1.mix&quot;</span>, <span class="st">&quot;A5.mix&quot;</span>, <span class="st">&quot;WB1145&quot;</span>, <span class="st">&quot;A6.mix&quot;</span>, <span class="st">&quot;A1.mix&quot;</span>)</span></code></pre></div>
<p>The betas generated earlier from openSesame are compressed into consensus vectors for each of the indicated cell types:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a>consensus_vector_names =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;NK&quot;</span>, <span class="st">&quot;Mono&quot;</span>, <span class="st">&quot;Tc&quot;</span>, <span class="st">&quot;Neu&quot;</span>, <span class="st">&quot;Th&quot;</span>, <span class="st">&quot;B&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>series_matrix_file &lt;-<span class="st">  </span><span class="kw">system.file</span>(<span class="st">&quot;extdata/series_matrix.rds&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;sesameData&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a>consensus_vector &lt;-<span class="st"> </span><span class="kw">betas_to_consensus_vector</span>(reference, series_matrix_file, betas, consensus_vector_names)</span></code></pre></div>
<div id="why-are-certain-probes-removed" class="section level2">
<h2>Why are certain probes removed?</h2>
<p>The openSesame pipeline sometimes replaces beta values with NA because of certain probe masking in the pipeline. These are masked because of their low quality during the Illumina assay.</p>
<p>Before the consensus vectors are built for each of the specified groups, the beta values are fit to a different bimodal distribution for each cell type. Several considerations about this distribution can explain the removal of certain probes. First, probes located on sex chromosomes are removed prior to creating the distributions because they have a disproportionate global methylation status. Also, probes that have a beta value outside of two standard deviations away from either of the two means in the distribution are labeled intermediates and removed.</p>
<p>In addition to these parameters, certain probes exhibit zero parsimony. Including them in tree reconstruction would be redundant as they do not add to the tree topology. Therefore, they are also removed.</p>
</div>
</div>
<div id="reconstructing-a-tree-using-the-icce-pipeline" class="section level1">
<h1>Reconstructing a tree using the ICCE Pipeline</h1>
<p>A tree can be constructed using the UPGMA algorithm with the following:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">upgma_tree</span>(consensus_vector)</span></code></pre></div>
<p>To load a known tree into the internal format of the pipeline, please use:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a>tree &lt;-<span class="st"> </span><span class="kw">load_tree</span>()</span></code></pre></div>
<p>Using the consensus vectors, the internal states of the tree can be reconstructed in the following way:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">build_icceTree</span>(consensus_vector, tree)</span></code></pre></div>
<div id="how-are-internal-states-reconstructed" class="section level2">
<h2>How are internal states reconstructed?</h2>
<p>Using a recursive implementation of Fitch’s algorithm, all of the internal states are inferred based off of the nodes of the tree and assigned as either a single state (methylated or unmethylated) or as an ambiguous state (methylated and unmethylated). In a second pass, those internal ambiguous states are resolved by inheriting parental unambiguous states to children.</p>
</div>
<div id="what-is-the-structure-of-an-iccetree-object" class="section level2">
<h2>What is the structure of an icceTree object?</h2>
<p>An icceTree object is composed of three different structures. The first structure is a n probe by m node data frame (prone-node matrix) that houses the methylation states of all of the nodes on the tree across all probes. A methylated status is assigned 1, unmethylated is 0, and ambiguous is 0.5. The second structure is also a data frame of the same size. This probe-node matrix houses p-values for each of the nodes across all probes. These are used for iterative reconstruction of the cellular tree when more and more data is added. the third structure is a phylogenetic tree structure from ape.</p>
</div>
</div>
<div id="editing-a-tree-using-the-icce-package" class="section level1">
<h1>Editing a tree using the ICCE package</h1>
<p>Data can be added to either one of the two probe-node matrices through:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">add_data</span>(icceTree, <span class="dt">n_new=</span><span class="dv">12</span>, <span class="dt">d=</span>icceTree icceTree<span class="op">$</span>probe_node_matrix[<span class="st">&#39;6&#39;</span>], <span class="dt">matrix=</span><span class="dv">0</span>)</span></code></pre></div>
<p>where matrix is a single value indicating to which probe-node matrix data will be added (0 == methyl states, 1 == p-values).</p>
<p>Either probe-node matrix can be relabeled with the following:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">relabel_node_matrix</span>(icceTree, <span class="dt">n_old=</span><span class="dv">7</span>, <span class="dt">old_new=</span><span class="dv">15</span>, <span class="dt">matrix=</span><span class="dv">2</span>)</span></code></pre></div>
<p>where matrix is a single value indicating to which probe-node matrix data will be added (0 == methyl states, 1 == p-values, 2 == both).</p>
<p>The phylo tree stored in the icceTree data structure can also be relabeled:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">relabel_tree</span>(icceTree, <span class="dt">n_old=</span><span class="dv">7</span>, <span class="dt">n_new=</span><span class="dv">15</span>)</span></code></pre></div>
<p>Certain parameters about the tree structure can also be obtained, such as a list of leaves, the number of leaves, the total number of nodes, and the number of internal nodes:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true"></a>leaves &lt;-<span class="st"> </span><span class="kw">get_leaves</span>(icceTree)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true"></a>number_of_leaves =<span class="st"> </span><span class="kw">nleaves</span>(icceTree)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true"></a>number_of_nodes =<span class="st"> </span><span class="kw">nnodes</span>(icceTree)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true"></a>number_of_internal_nodes =<span class="st"> </span><span class="kw">ninternal_nodes</span>(icceTree)</span></code></pre></div>
<p>Also the root of the tree can be obtained with:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true"></a>root &lt;-<span class="st"> </span><span class="kw">get_root</span>(icceTree)</span></code></pre></div>
<p>A tree can be grown from any current leaf on the tree structure with:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">grow_off_leaf</span>(icceTree, <span class="dt">leaf=</span><span class="dv">1</span>, <span class="dt">node_names=</span><span class="kw">c</span>(<span class="st">&#39;Fibroblasts&#39;</span>, <span class="st">&#39;Fibrocytes&#39;</span>))</span></code></pre></div>
<p>where node_names contains the labels for the two new leaves add at the first node.</p>
<p>The tree can also be cut at any node using:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true"></a>icceTree &lt;-<span class="st"> </span><span class="kw">remove_branches_from_node</span>(icceTree, <span class="dt">node=</span><span class="dv">8</span>)</span></code></pre></div>
<p>where the 8th node will become a new leaf and all of its descendants will be removed from the tree structure. Descendants are also automatically removed from both the methyl probe-node matrix and the p-value probe-node matrix.</p>
</div>
<div id="analyzing-the-reconstructed-tree-using-icce-pipeline" class="section level1">
<h1>Analyzing the reconstructed tree using ICCE pipeline</h1>
<p>Differentially methylated probes on each branch can be obtained using:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true"></a>probe_information &lt;-<span class="st"> </span><span class="kw">generate_branch_changes</span>(icceTree)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true"></a>relevant_probes &lt;-<span class="st"> </span>probe_information<span class="op">$</span>relevant_probes</span></code></pre></div>
<p>The probe_information variable also has three other structures in it: a node label list for the total number of changes in methylation, number of gains in methylation, and the number of losses in methylation along each branch on the tree:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true"></a>total_changes_label &lt;-<span class="st"> </span>probe_information<span class="op">$</span>total_changes_label</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true"></a>meth_gains_label &lt;-<span class="st"> </span>probe_information<span class="op">$</span>meth_gains_label</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true"></a>meth_losses_label &lt;-<span class="st"> </span>probe_information<span class="op">$</span>meth_losses_label</span></code></pre></div>
<p>Using these relevant probes, corresponding differentially methylated genes can be obtained with:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true"></a>relevant_genes &lt;-<span class="st"> </span><span class="kw">get_relevant_genes</span>(icceTree, relevant_probes, reference)</span></code></pre></div>
<p>One powerful tool of this package is the ability to track the differentiation on the tree of probes along a selected gene:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true"></a>gene_information &lt;-<span class="st"> </span><span class="kw">track_gene</span>(icceTree, reference, <span class="dt">gene=</span><span class="st">&#39;CD81&#39;</span>)</span></code></pre></div>
<p>where the output are two node label lists with the proportion of methylated internal nodes and leaves at each of the three states (methylated, unmethylated, and ambiguous).</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true"></a>thermo_prop_internal_nodes =<span class="st"> </span>gene_information<span class="op">$</span>thermo_prop_internal_nodes</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true"></a>thermo_prop_leaves =<span class="st"> </span>gene_information<span class="op">$</span>thermo_prop_leaves</span></code></pre></div>
</div>
<div id="visualizing-trees-using-the-icce-package" class="section level1">
<h1>Visualizing trees using the ICCE package</h1>
<p>A reconstructed tree can be visualized with the following command:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true"></a><span class="kw">show_tree</span>(<span class="dt">filename=</span><span class="st">&#39;tree.pdf&#39;</span>, <span class="dt">tree=</span>icceTree<span class="op">$</span>tree, <span class="dt">title=</span><span class="st">&#39;Hematopoietic Tree&#39;</span>, <span class="dt">edge_labels=</span>meth_gains_label)</span></code></pre></div>
<p>where edge_labels is an optional parameter.</p>
<p>The edge labels generated for a specific gene from the track_gene function can be visualized as well:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true"></a><span class="kw">show_gene_track</span>(<span class="st">&#39;tree_CD81&#39;</span>, <span class="dt">tree=</span>icceTree<span class="op">$</span>tree, <span class="dt">gene=</span><span class="st">&#39;CD81&#39;</span>, <span class="dt">thermo_prop_internal_nodes=</span>thermo_prop_internal_nodes, <span class="dt">thermo_prop_leaves=</span>thermo_prop_leaves)</span></code></pre></div>
<p>Another useful plot is a heat map of probes with high variance:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true"></a><span class="cf">in</span> progress...</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true"></a>head_map &lt;-<span class="st"> </span><span class="kw">show_heatmap</span>()</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
